packages/better-php-doc-parser/tests/PhpDocInfo/PhpDocInfoPrinter/FixtureCallable/callable_mixed.txt

```diff
@@ -1 +1 @@
-/** @var callable(array<mixed>) */
+/** @var callable(array<mixed>):mixed */
```

Comment:
Please keep these 2 and add 2 new extra files, so it's clear there is no change in behavior

Reply:
But this behavior it's what we need. We need to disallow typed callable without return type via https://github.com/phpstan/phpstan/issues/3008

Reply:
How would you explain people a rule that changes their phpdocs?

Reply:
IMO `mixed` means same `no return type`. So this change just improve `callable` definition.

Currently rector changes my phpdocs by remove mixed, so who would you explain me this change ;-)

Reply:
That's just moving problem from to right.
The solution is to respect both cases.

Reply:
This change was introduced there https://github.com/rectorphp/rector/pull/2884 right?

Reply:
@ondrejmirtes maybe you can add something to discussion? Your voice is important because currently this trigger error in phpstan https://github.com/phpstan/phpstan/issues/3008

Reply:
@snapshotpl Probably. I have focused on many issues since.

Reply:
`callable(X)` without return typehint in PHPDoc was never supported, because it's ambiguous for the AST parser. Better to use the `callable(X): mixed` syntax.

Is it `IdentifierTypeNode` with `(X)` description, or is it `CallableTypeNode`? There's no way to tell. I added some tests with the current behaviour based on the issue https://github.com/phpstan/phpdoc-parser/commit/e2b89c5119b378518c259ea4d5ff16c662281f84

Reply:
@TomasVotruba what are you thing now?

This change is introduced by my request, so probably nobody cares about "revert" this behavior.

If it's BC break you can start with 0.8, but IMO callable, if typed, should always contains return type.

Reply:
Format must be preserved. It's the most discouraging to see no change yet 200 changed files